realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -620,44 +622,136 @@ public static AllTypes copyOrUpdate(Realm realm, AllTypes object, boolean update
         }
     }
 
-    public static AllTypes copy(Realm realm, AllTypes newObject, boolean update, Map<RealmObject,RealmObjectProxy> cache) {
-        AllTypes realmObject = realm.createObject(AllTypes.class, newObject.getColumnString());
-        cache.put(newObject, (RealmObjectProxy) realmObject);
-        realmObject.setColumnString(newObject.getColumnString());
-        realmObject.setColumnLong(newObject.getColumnLong());
-        realmObject.setColumnFloat(newObject.getColumnFloat());
-        realmObject.setColumnDouble(newObject.getColumnDouble());
-        realmObject.setColumnBoolean(newObject.isColumnBoolean());
-        realmObject.setColumnDate(newObject.getColumnDate());
-        realmObject.setColumnBinary(newObject.getColumnBinary());
+    public static AllTypes copy(Realm realm, AllTypes from, boolean update, Map<RealmObject,RealmObjectProxy> cache) {
+        final boolean isStandalone = !(from instanceof AllTypesRealmProxy);
+        Class<? extends AllTypes> clazz;
+        Field field = null;
+        if (isStandalone) {
+            clazz = from.getClass();
+        } else {
+            clazz = null;
+        }
+        AllTypesRealmProxy to;
 
-        some.test.AllTypes columnObjectObj = newObject.getColumnObject();
-        if (columnObjectObj != null) {
-            some.test.AllTypes cachecolumnObject = (some.test.AllTypes) cache.get(columnObjectObj);
-            if (cachecolumnObject != null) {
-                realmObject.setColumnObject(cachecolumnObject);
-            } else {
-                realmObject.setColumnObject(AllTypesRealmProxy.copyOrUpdate(realm, columnObjectObj, update, cache));
+        if (isStandalone) {
+            try {
+                field = clazz.getDeclaredField("columnString");
+            } catch (NoSuchFieldException e) {
+                throw new RealmException(e.getMessage());
+            }
+            field.setAccessible(true);
+            Object primaryKey;
+            try {
+                primaryKey = field.get(from);
+            } catch (IllegalAccessException e) {
+                throw new RealmException(e.getMessage());
             }
+            to = (AllTypesRealmProxy) realm.createObject(AllTypes.class, primaryKey);
         } else {
-            realmObject.setColumnObject(null);
+            to = (AllTypesRealmProxy) realm.createObject(AllTypes.class, ((AllTypesRealmProxy) from).realmGetter$columnString());
```

Comment:
Is it possible just have `to = (AllTypesRealmProxy) realm.createObject(AllTypes.class, ((AllTypesRealmProxy) from).realmGetter$columnString())` even for standalone object just change it to `to = ealm.createObject(AllTypes.class, from.realmGetter$columnString())` during byte code manipulating?